src/main/java/com/reservation/hall/domain/Hall.java

```diff
@@ -0,0 +1,94 @@
+package com.reservation.hall.domain;
+
+import com.reservation.hall.domain.exception.InvalidHallStatusChangeException;
+import lombok.EqualsAndHashCode;
+
+import java.util.Objects;
+import java.util.Set;
+
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+public class Hall {
+    @EqualsAndHashCode.Include
+    private final HallId id;
+    private HallName name;
+    private Capacity capacity;
+    private Set<Equipment> equipment ;
+    private HallStatus status;
+
+    private Hall(HallId id, HallName name, Capacity capacity, Set<Equipment> equipment) {
+        this.id = Objects.requireNonNull(id);
+        this.name = Objects.requireNonNull(name);
+        this.capacity = Objects.requireNonNull(capacity);
+        this.equipment = Set.copyOf(Objects.requireNonNull(equipment));
+        this.status = HallStatus.ACTIVE;
+    }
+
+    public static Hall create(HallId id, HallName name, Capacity capacity, Set<Equipment> equipment) {
+        return new Hall(id, name, capacity, equipment);
+    }
+
+    public HallId getHallId() {
+        return this.id;
+    }
+
+    public void rename(HallName name) {
+        this.name = Objects.requireNonNull(name);
+    }
+
+    public void resize(Capacity capacity) {
+        this.capacity = Objects.requireNonNull(capacity);
+    }
+
+    public void updateEquipment(Set<Equipment> equipment) {
+        this.equipment = Set.copyOf(Objects.requireNonNull(equipment));
+    }
+
+    public void activate() {
+        if(this.status == HallStatus.ACTIVE) {
+            throw new InvalidHallStatusChangeException("Hall is already active");
+        }
+        this.status = HallStatus.ACTIVE;
+    }
```

Comment:
Poprawić możemy jeszcze czytelność i uniknąć duplikowania kodu
```suggestion
    public void activate() {
        if (isActive()) {
            throw new InvalidHallStatusChangeException("Hall is already active");
        }
        this.status = HallStatus.ACTIVE;
}
```